projects/addon-charts/components/line-chart/line-chart-hint.directive.ts

```diff
@@ -26,12 +26,20 @@ import {combineLatest, Observable} from 'rxjs';
 import {distinctUntilChanged, filter, map, startWith, takeUntil} from 'rxjs/operators';
 
 import {TuiLineChartComponent} from './line-chart.component';
+import {AbstractTuiLineChartHintToken} from './line-chart-hint.token';
 
 @Directive({
     selector: `[tuiLineChartHint]`,
-    providers: [TuiDestroyService, TuiHoveredService],
+    providers: [
+        TuiDestroyService,
+        TuiHoveredService,
+        {provide: AbstractTuiLineChartHintToken, useExisting: TuiLineChartHintDirective},
+    ],
 })
-export class TuiLineChartHintDirective implements AfterContentInit {
+export class TuiLineChartHintDirective
+    extends AbstractTuiLineChartHintToken
```

Comment:
make it sense to extend abstract class and add super() if abstract class is empty?

Reply:
good catch